public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx

```diff
@@ -196,3 +233,72 @@ export function getPanelLinksBehavior(panel: PanelModel) {
     panelLinksMenu.setState({ links });
   };
 }
+
+function createExtensionContext(panel: VizPanel, dashboard: DashboardScene): PluginExtensionPanelContext {
+  const timeRange = sceneGraph.getTimeRange(panel);
+  let queryRunner = panel.state.$data;
+  let targets: DataQuery[] = [];
+  const id = getPanelIdForVizPanel(panel);
+
+  if (queryRunner instanceof SceneDataTransformer) {
+    queryRunner = queryRunner.state.$data;
+  }
+
+  if (queryRunner instanceof SceneQueryRunner) {
+    targets = queryRunner.state.queries;
+  }
+
+  if (queryRunner instanceof ShareQueryDataProvider) {
+    targets = [queryRunner.state.query];
+  }
+
+  let scopedVars = {};
+
+  // Handle panel repeats scenario
+  if (panel.state.$variables && panel.state.$variables instanceof SceneVariableSet) {
+    panel.state.$variables.state.variables.forEach((variable) => {
+      if (variable instanceof LocalValueVariable) {
```

Comment:
I don't think we need this check here.

Reply:
It's actually helpful as LocalValueVariable guarantees `getValueText` implementation and we do not provide any other variable types to that set at the moment.